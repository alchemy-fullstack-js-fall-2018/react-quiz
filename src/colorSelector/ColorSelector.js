import React from 'react';

export default class ColorSelector extends Component {

  state = {
    fontColor: 'red',
    backgroundColor: ''
  }

  onChange = ({ target }) => {
    // setState in async. It waits for all components to update
    this.setState({ [target.name]: target.value });
  };

  render() {
    const { fontColor, textArea, backgroundColor } = this.state;

    const fontOptions = [ 'black', 'white', 'blue', 'red', 'yellow'].map(font => {
      return <option key={font} value={font}>{font}</option>;
    });

    const backgroundColor = [ 'black', 'white', 'blue', 'red', 'yellow'].map(backColors => {
      return <option key={backColors} value={backColors}>{backColors}</option>;
    });


    const textStyles = () => {
      return {
        fontColor: fontColor,
        backgroundColor: backgroundColor
      }
    }

    return (
      <Fragment>
        <h1>Select Fonts</h1>

          <div>
            <label htmlFor="font">Select Font Color </label>
            <select name="font" defaultValue={font} onChange={this.onChange}>
              {fontOptions}
            </select>
          </div>

          <div>
            <label htmlFor="backColors">Select color background </label>
            <select name="backColors" defaultValue={backColors} onChange={this.onChange}>
              {backgroundColor}
            </select>
          </div>

          <div>
            <label htmlFor="textArea">Enter Text here</label>
            <input name="textArea" type="text" value={textArea} onChange={this.onChange} />
          </div>
          <div style={textStyles()}>
              {textArea}
            </div>
      </Fragment>
    );
}
