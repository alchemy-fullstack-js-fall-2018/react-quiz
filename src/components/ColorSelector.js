import React from 'react';

export default function ColorSelector({
  fontColor,
  backgroundColor,
  onChange
}) {
  return (
    <div>
      <label name="fontColor">Choose a font color:</label>
      <input
        name="fontColor"
        type="color"
        value={fontColor}
        onChange={onChange}
      />
      <label name="backgroundColor">Choose a background color:</label>
      <input
        name="backgroundColor"
        type="color"
        value={backgroundColor}
        onChange={onChange}
      />
    </div>
  );
}
