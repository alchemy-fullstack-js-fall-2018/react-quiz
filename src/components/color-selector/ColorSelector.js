import React, { Fragment } from 'react';

const ColorSelector = ({ textColor, backgroundColor, handleChange }) => {

  const textColorOptions = ['black', 'white', 'blue', 'red', 'yellow'].map(color => {
    return <option key={color} value={color}>{color}</option>;
  });
  const backgroundColorOptions = ['black', 'white', 'blue', 'red', 'yellow'].map(color => {
    return <option key={color} value={color}>{color}</option>;
  });
  return (
    <Fragment>
      <select name={textColor} defaultValue={textColor} onChange={handleChange}>
        {textColorOptions}
      </select>
      <select name={backgroundColor} defaultValue={backgroundColor} onChange={handleChange}>
        {backgroundColorOptions}
      </select>

    </Fragment>
  );
};

export default ColorSelector;
