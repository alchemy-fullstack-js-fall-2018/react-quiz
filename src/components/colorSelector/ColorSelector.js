import React, { Fragment } from 'react';
import PropTypes from 'prop-types';

function ColorSelector({ artist, works }) {

  return (
    <Fragment>
      <input
        type="color"
      ></input>

    </Fragment>
  );
}

ColorSelector.propTypes = {
  // note: PropTypes.object.isRequired
};

export default ColorSelector;
