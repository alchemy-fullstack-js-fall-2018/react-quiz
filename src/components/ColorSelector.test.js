import React from 'react';
import ColorSelector from './ColorSelector';
import render from 'react-test-renderer';

describe('Color Selector Test', () => {
  test('ColorSelector snapshot', () => {
    expect(render
      .create(<ColorSelector />)
      .toJSON()).toMatchSnapshot();
  });
});
