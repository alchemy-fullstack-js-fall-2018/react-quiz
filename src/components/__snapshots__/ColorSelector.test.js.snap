// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Color Selector Test ColorSelector snapshot 1`] = `
<div>
  <label
    name="fontColor"
  >
    Choose a font color:
  </label>
  <input
    name="fontColor"
    type="color"
  />
  <label
    name="backgroundColor"
  >
    Choose a background color:
  </label>
  <input
    name="backgroundColor"
    type="color"
  />
</div>
`;
